src/components/reservations/ReservationsCalendar/ReservationForm.tsx

```diff
@@ -0,0 +1,204 @@
+'use client';
+
+import { Button } from '@/components/ui/Button';
+import { DateTimePicker } from '@/components/ui/DateTimePicker';
+import {
+  Form,
+  FormControl,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  useForm,
+} from '@/components/ui/Form';
+import { Input } from '@/components/ui/Input';
+import { useTranslations } from 'next-intl';
+import type React from 'react';
+import { type FormEvent, useEffect, useState } from 'react';
+import { z } from 'zod';
+
+type Reservation = {
+  navn: string;
+  mobilNr: string;
+  email: string;
+  start: Date | string;
+  end: Date | string;
+};
+
+type ReservationFormProps = {
+  start: Date;
+  end: Date;
+  mode: 'create' | 'view';
+  onSubmit: (data: Reservation) => void;
+  onCancel: () => void;
+  onDelete?: () => void;
+  defaultValues?: Omit<Reservation, 'start' | 'end'>;
+};
+
+export function ReservationForm({
+  start,
+  end,
+  mode,
+  onSubmit,
+  onCancel,
+  onDelete,
+  defaultValues,
+}: ReservationFormProps) {
+  const t = useTranslations('reservations');
+  const formSchema = z.object({
+    navn: z.string().min(1, t('form.formSchema.nameReq')),
+    mobilNr: z.string().min(1, t('form.formSchema.phoneNrReq')),
+    email: z.string().min(1, t('form.formSchema.emailReq')),
+    start: z.date().or(z.string()),
+    end: z.date().or(z.string()),
+  });
+  const [startDate, setStartDate] = useState<Date>(start);
+  const [endDate, setEndDate] = useState<Date>(end);
+  const form = useForm(formSchema, {
+    defaultValues: {
+      navn: defaultValues?.navn ?? '',
+      mobilNr: defaultValues?.mobilNr ?? '',
+      email: defaultValues?.email ?? '',
+      start: start || new Date(),
+      end: end || new Date(),
+    },
+  });
+
+  /** I tilfelle personen endrer tiden igjen i formen */
+  useEffect(() => {
+    setStartDate(start);
+    setEndDate(end);
+  }, [start, end]);
+
+  /** FormEvent<HTMLFormElement>, fordi onSubmit tar inn den her typen */
+  function handleFormSubmit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    try {
+      onSubmit({
+        navn: form.getFieldValue('navn'),
+        mobilNr: form.getFieldValue('mobilNr'),
+        email: form.getFieldValue('email'),
+        start: startDate.toISOString(),
+        end: endDate.toISOString(),
+      });
+    } catch (error) {
+      // noe toast/alert her
+    }
+  }
+
+  return (
+    <Form onSubmit={handleFormSubmit} className='w-full space-y-4 p-4'>
+      <form.Field name='navn'>
+        {(field) => (
+          <FormItem errors={[]}>
```

Comment:
in the old solution you would set errors like this so they are displayed when the form is validated
```suggestion
          <FormItem errors={field.state.meta.errors}>
```

Reply:
It is done automatically in Form v1